### PR TITLE
[Groups] Implimented groups individual page and routing between them

### PR DESCRIPTION
**Pull request by eiffelwong1**
Issue: #??

## Summary (1-2 sentences)

Created new individual group pages, Additionally added routing between 
- groups page <-> individual group page
- groups page <-> new groups page
- footer <-> login, signup, groups

## Details (reason and description of the changes)

new features, who doesn't; like it? :D

### Challenges (what problems did I face?) (optional)

- remix routing defaults to nested routing, which is a bit tricky until we figure out the syntax of it.
- the db needs a await to actually store things

### Solution (how did I fix the problems?) (optional)

elbow grease, Tony's knowledge and Google

### Sources (links that helped to resolve these challenges.) (optional)

https://remix.run/docs/en/main/file-conventions/routes#nested-urls-without-layout-nesting

## Screenshots/Recordings (if applicable)

main features:

https://github.com/social-plan-it/plan-it-social-web/assets/29219684/e6d21a15-25ef-42bf-a37c-81467241036a

error boundary:

https://github.com/social-plan-it/plan-it-social-web/assets/29219684/00b43f4f-629c-4f26-9b1c-b2a0256260bd

 

 

## Co-authored-by:
Tony Dang @tonydangblog
Elizabeth

### CODE REVIEW

File: app/components/ui/forms.tsx [hunk 1]
```diff
@@ -37,10 +38,23 @@ export function TextArea({ label, ...props }: TextAreaProps) {
...
   return (
-    <button {...props} className="border-2 w-24 bg-red-700 text-white rounded-lg" type="submit">
-      Create
+    <button {...props} className="border-2 bg-red-700 text-white rounded-lg" type="submit">
+      Create {children}
     </button>
   );
 }
+
+type LinkButtonProps = {
```
Comment (eiffelwong1) [inline app/components/ui/forms.tsx:49]: we found this method to put `to` into the prop type, but it would be better if there's a react prop type for Link

**Review by eiffelwong1 — COMMENTED** [review]
init comments

File: app/routes/groups_.new.tsx [hunk 2]
```diff
@@ -10,41 +31,43 @@ export default function GroupNew() {
           <H1>Create New Group</H1>
           <H2>Your Community Starts Here</H2>
           <Card>
-            <div className="flex pt-4 w-full">
-              <div className="w-1/2">
-                <div className="w-1/2 pb-4">
-                  <Input label="Group Name:" name="groupName" required />
+            <Form method="post">
```
Comment (eiffelwong1) [inline app/routes/groups_.new.tsx:34]: the only change in this file is adding in Form

File: app/routes/groups_.$groupId.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import type { LoaderFunction } from '@remix-run/node';
+import { useParams, useLoaderData, Link } from '@remix-run/react';
+import { db } from '~/modules/database/db.server';
+import { json } from '@remix-run/node';
+
+export const loader: LoaderFunction = async ({ params }) => {
+  const group = await db.group.findFirstOrThrow({ where: { id: params.groupId } });
```
[the remark this answers is not in the document]
Comment (shami-dev) [reply app/routes/groups_.$groupId.tsx:7]: findFirstOrThrow :cool:

File: app/components/layout/footer.tsx [hunk 4]
```diff
@@ -25,12 +25,12 @@ export default function Footer() {
           <ul className="mb-8">
             <li className="text-lg">Your Account</li>
             <li className="pb-1">
-              <a className="text-gray-400   font-thin" href="/">
+              <a className="text-gray-400   font-thin" href="/signup">
```
Comment (andrelandgraf) [inline app/components/layout/footer.tsx:28]: We could also use Remix's `Link` component here (next)! (Not a blocker!)

File: app/components/ui/forms.tsx [hunk 5]
```diff
@@ -37,10 +38,23 @@ export function TextArea({ label, ...props }: TextAreaProps) {
 
 type ButtonProps = ButtonHTMLAttributes<HTMLButtonElement>;
 
-export function Button({ ...props }: ButtonProps) {
+export function Button({ children, ...props }: ButtonProps) {
   return (
-    <button {...props} className="border-2 w-24 bg-red-700 text-white rounded-lg" type="submit">
-      Create
+    <button {...props} className="border-2 bg-red-700 text-white rounded-lg" type="submit">
+      Create {children}
```
Comment (andrelandgraf) [inline app/components/ui/forms.tsx:44]: Let's remove the static string altogether! So we can have a button with any text!

```suggestion
      {children}
```

File: app/components/ui/forms.tsx [hunk 6]
```diff
@@ -37,10 +38,23 @@ export function TextArea({ label, ...props }: TextAreaProps) {
...
+
+type LinkButtonProps = {
+  children: ReactNode;
+  to: string;
+};
+
+export function LinkButton({ to, children, ...props }: LinkButtonProps) {
+  return (
+    <Link {...props} to={to} className="border-2 bg-red-700 text-white rounded-lg p-3 w-full text-center" type="submit">
+      Create {children}
```
Comment (andrelandgraf) [inline app/components/ui/forms.tsx:57]: Same here! Let's keep it super generic!

```suggestion
      {children}
```

File: app/components/ui/forms.tsx (hunk 1 above)
Comment (andrelandgraf) [reply app/components/ui/forms.tsx:49]: I think Remix exports a `LinkProps` type or similar that we could use here!

File: app/routes/groups.tsx [hunk 7]
```diff
@@ -1,15 +1,26 @@
...
-        })}
-      </ul>
-    </Card>
+    <>
+      <Card>
+        <ul className="max-w-md space-y-1 text-gray-500 list-disc list-inside dark:text-gray-400">
+          {groups?.map((group) => {
+            return (
+              <li key={group.id}>
+                <Link to={group.id}>{group.name}</Link>
```
Comment (andrelandgraf) [inline app/routes/groups.tsx:15]: Personal preference: Maybe let's use absolute paths only. So if we ever move this code into a reusable component or into a different route, it still works!

**Review by andrelandgraf — APPROVED** [review]
Great work!

**Review by andrelandgraf — APPROVED** [review]
Great work!

**Review by tonydangblog — APPROVED** [review]

**Pull request merged**